modules/core/src/main/scala/lucuma/gen/Generator.scala

```diff
@@ -3,50 +3,96 @@
 
 package lucuma.gen
 
+import cats.Monad
 import cats.effect.Sync
-import fs2.Stream
-import lucuma.odb.api.model.{AtomModel, StepModel}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import lucuma.core.`enum`.Instrument
+import lucuma.odb.api.model.{ExecutionModel, Sequence, SequenceModel}
 
 /**
  * A sequence generator for the given static `S` and dynamic `D` types for a
  * particular observing mode.
  *
+ * @tparam F effect type
  * @tparam S type of static (unchanging) configuration data
  * @tparam D type of dynamic (potentially changing from step to step) configuration
  */
-trait Generator[S, D] {
+trait Generator[F[_], S, D] { self =>
 
+  def instrument: Instrument
+
+  /**
+   * Static, unchanging, configuration for this observing mode.
+   */
   def static: S
 
   /**
-   * Generates a full acquisition sequence, stopping when the given `acquired`
-   * computation produces a `true` value.
-   *
-   * @param acquired a computation that determines when the target is acquired
+   * Generates the acquisition sequence expected for the remainder of the
+   * acquisition.
    */
-  def acquisition[F[_]: Sync](
-    acquired: F[Boolean]
-  ): Stream[F, AtomModel[StepModel[D]]]
+  def acquisition(
+    recordedSteps: List[RecordedStep[D]]
+  ): F[Sequence[D]]
 
   /**
-   * Generates a re-acquisition sequence, stopping only when the given
-   * `acquired` computation produces a `true` value.
-   *
-   * @param acquired a computation that determines when the target is reacquired
+   * Generates the science sequence expected for the remainder of the
+   * observation.
    */
-  def reacquisition[F[_]: Sync](
-    acquired: F[Boolean]
-  ): Stream[F, AtomModel[StepModel[D]]]
+  def science(
+    recordedSteps: List[RecordedStep[D]]
+  ): F[Sequence[D]]
+
+
+}
+
+object Generator {
 
   /**
-   * Generates the full science sequence, stopping when the given `observed`
-   * computation produces a `true` value (e.g., when the required S/N has been
-   * reached).
-   *
-   * @param observed a computation that decides when the observation is done
+   * Runs the Generator to make the execution config with its static config,
+   * acquisition sequence, and science sequence.
    */
-  def science[F[_]: Sync](
-    observed: F[Boolean]
-  ): Stream[F, AtomModel[StepModel[D]]]
+  def run[F[_]: Monad, S, D](
```

Comment:
Can this be a method in `Generator` ?

Reply:
Yes, I wanted it to be there but was stymied.  How can I stipulate that `F[_]` have a `Monad` if I add move it to the instance?

Reply:
We can have an `implicit F: Monad[F]` parameter in the method itself.

Reply:
Gah, of course!  Thanks, done.